packit_service/worker/handlers/copr.py

```diff
@@ -479,3 +504,160 @@ def handle_testing_farm(self):
                         "build_id": self.build.id,
                     },
                 ).apply_async()
+
+
+class ScanHelper:
+    def __init__(
+        self, copr_build_helper: CoprBuildJobHelper, build: CoprBuildTargetModel
+    ):
+        self.build = build
+        self.copr_build_helper = copr_build_helper
+
+    def handle_scan(self):
+        """
+        Try to find a job that can provide the base SRPM,
+        download both SRPM and base SRPM and trigger the scan in OpenScanHub.
+        """
+        base_build_job = self.find_base_build_job()
+        if not base_build_job:
+            logger.debug("No base build job needed for diff scan found in the config.")
+            return
+
+        logger.info("Preparing to trigger scan in OpenScanHub...")
+
+        base_srpm_model = self.get_base_srpm_model(base_build_job)
+
+        if not base_srpm_model:
+            return
+
+        srpm_model = self.build.get_srpm_build()
+
+        with tempfile.TemporaryDirectory() as directory:
+            paths = self.download_srpms(directory, base_srpm_model, srpm_model)
+            if not paths:
+                return
+
+            build_dashboard_url = get_copr_build_info_url(self.build.id)
+
+            output = self.copr_build_helper.api.run_osh_build(
+                srpm_path=paths[1],
+                base_srpm=paths[0],
+                comment=f"Submitted via Packit Service for {build_dashboard_url}.",
+            )
+
+            if not output:
+                logger.debug("Something went wrong, skipping the reporting.")
+                return
+
+            logger.info(f"Scan submitted successfully, output: {output}")
+
+            response = json.loads(output)
+            url = response.get("url")
+
+            if not url:
+                logger.debug("It was not possible to get the URL from the response.")
+                return
+
+            self.copr_build_helper._report(
+                state=BaseCommitStatus.success,
+                description=(
+                    "Scan in OpenScanHub submitted succesfully. Check the URL for more details."
+                ),
+                url=url,
+                check_names=["osh-diff-scan:fedora-rawhide-x86_64"],
+                markdown_content=(
+                    "This is an experimental feature. Once the scan finishes, you can see the "
+                    "newly introduced defects in the `added.html` in `Logs`. "
+                    "You can disable the scanning in your configuration by "
+                    "setting `osh_diff_scan_after_copr_build` to `False`. For more information, "
```

Comment:
Just a nitpick, but in the docs we tend to use the lowercase form:

```suggestion
                    "setting `osh_diff_scan_after_copr_build` to `false`. For more information, "
```